### DISCUSSION TITLE
EagerLoad doesn´t bring selected fields, only all of them.

### DISCUSSION THREAD

**Discussion by mvrochar** [adonisjs/core#2265] [Help] [answered]
I have a eagerLoad using .with method, where it brings all the field from another table, wich are not necessary. So I want to select only the fields I need, the problem is, when I use the callback, like this: `   .with('manager', (builder) => {
           builder.select('numcad', 'nomfun')
          })`, it just returns null.

What would be the correct way to just bring the fields I need?

Thank in advance

**Comment by JaconsMorais** [comment, marked as answer]
Hi, you have two ways to do this.
1# you can use the hidden() getter on your model, but then all your loads will be without the hidden fields.
2# you can use setVisible nested with query() statement, like [this](https://adonisjs.com/docs/4.1/lucid#_setvisiblesethidden)

So, in your example you can use:

`
.with('manager', (builder) => { builder.setVisible(['numcad', 'nomfun']) })
`

**Reply by mvrochar** [reply]
It worked! Thank you very much!!
